### PR TITLE
feat: use new performance apis for search campaigns

### PR DESCRIPTION
**Pull request by tackley**
Use the updated performance apis that do server-based aggregation, so the screens do not overwhelm the browser on large campaigns.

To introduce this incrementally, this is being added for search campaigns only initially, with the intention of subsequently switching other campaign types to use this in due course.

Although the original intention was to keep the UI the same, this required a lot of throw-away work, so I've brought in the experience of what we now do on our internal dashboards.

Outstanding steps:
- [x] re-instate date filtering
- [x] ad / adset breakdowns
- [x] review list of metrics

### CODE REVIEW

File: src/user/analytics/search/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    shortCaption: msg`CTR`,
+    tooltip: msg`The rate at which users clicked in correlation to their impressions`,
+    getValue: (metrics) => BigNumber(metrics.rates.clickThrough),
+    type: "rate",
+    color: colors[2],
+  },
+  {
+    id: "site-visit",
+    caption: msg`Site visits`,
+    tooltip: msg`Counted if the user clicks an ad and spends at least 5 seconds on the advertiser's website, with the website open in an active browser tab. The 10 seconds must be spent on the site after arriving by clicking the ad link, and the tab must remain open and active the entire time for the visit to count.`,
```
Comment (IanKrieger) [inline src/user/analytics/search/metrics.ts:62]:
```suggestion
    tooltip: msg`Counted if the user clicks an ad and spends at least 5 seconds on the advertiser's website, with the website open in an active browser tab. The 5 seconds must be spent on the site after arriving by clicking the ad link, and the tab must remain open and active the entire time for the visit to count.`,
```

Comment (tackley) [reply src/user/analytics/search/metrics.ts:62]: Actually this change isn't actually live yet so I'll leave it at 10 seconds.

File: src/locales/es.po [hunk 2]
```diff
@@ -612,10 +626,17 @@ msgid "Count conversions within X days of an impression"
...
 
 #: src/user/analytics/analyticsOverview/components/MetricSelect.tsx:46
-msgid "Counted if the user clicks an ad and spends at least 10 seconds on the advertiser's website, with the website open in an active browser tab. The 10 seconds must be spent on the site after arriving by clicking the ad link, and the tab must remain open and active the entire time for the visit to count."
+#: src/user/analytics/search/metrics.ts:62
+#~ msgid "Counted if the user clicks an ad and spends at least 10 seconds on the advertiser's website, with the website open in an active browser tab. The 10 seconds must be spent on the site after arriving by clicking the ad link, and the tab must remain open and active the entire time for the visit to count."
+#~ msgstr ""
+
+#: src/user/analytics/analyticsOverview/components/MetricSelect.tsx:46
+#: src/user/analytics/search/metrics.ts:62
+msgid "Counted if the user clicks an ad and spends at least 5 seconds on the advertiser's website, with the website open in an active browser tab. The 10 seconds must be spent on the site after arriving by clicking the ad link, and the tab must remain open and active the entire time for the visit to count."
```
Comment (iambrianfung) [inline src/locales/es.po:635]: i don't think this is live

Comment (tackley) [reply src/locales/es.po:635]: yup i was over-keen with that change, backing it out

**Pull request merged**